npush: init at 0.7

AndersonTorres opened the pull request:
Also, include a runner script to help copying level files before running the
game.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

legendofmiracles commented on pkgs/games/npush/run.nix:
```diff
@@ -0,0 +1,39 @@
+
+      exec "${npush}/bin/npush"
+    '';
+    executable = true;
+    destination = "/bin/run-npush";
```
These shouldn't be needed if you use the more idiomatic function

IvarWithoutBones commented on pkgs/games/npush/run.nix:
```diff
@@ -0,0 +1,39 @@
+}:
+
+let
+  runScript = writeTextFile {
+    name = "run-npush";
+    text = ''
+      #!${runtimeShell}
```
See https://github.com/NixOS/nixpkgs/pull/155263#pullrequestreview-855754612 for the reasoning

AndersonTorres replied:
As I have seen, this is not the expected syntax of `writeShellScriptBin`. 

I will fix this nonetheless

sikmir commented on pkgs/games/npush/default.nix:
```diff
@@ -0,0 +1,47 @@
+  version = "0.7";
+
+  src = fetchurl {
+    url = "https://downloads.sourceforge.net/project/npush/${pname}/${version}/${pname}-${version}.tgz";
```
```suggestion
    url = "mirror://sourceforge/project/npush/npush/${version}/npush-${version}.tgz";
```

AndersonTorres committed:
npush: init at 0.7

Also, adding a helper run-npush that takes care of preparing the game to run
with the official levelset.
